modules/compiler/targets/host/source/target.cpp

```diff
@@ -186,9 +186,21 @@ compiler::Result HostTarget::initWithBuiltins(
       triple = llvm::Triple("armv7-unknown-linux-gnueabihf-elf");
       break;
     case host::arch::AARCH64:
-      assert(host::os::LINUX == host_device_info.os &&
-             "AArch64 cross-compile only supports Linux");
-      triple = llvm::Triple("aarch64-linux-gnu-elf");
+      switch (host_device_info.os) {
+        case host::os::LINUX:
+          triple = llvm::Triple("aarch64-linux-gnu-elf");
+          break;
+        case host::os::MACOS:
+          assert(host_device_info.native &&
+                 "macOS cross-compile not supported");
+          // Our generated code does not meet Mach-O restrictions. Force ELF
```

Comment:
I think it would be helpful to say what the restriction is like in the PR description.

Reply:
Agreed. During a simple build, we get during the execution of `clc`:
```
LLVM ERROR: Global variable 'notes_global' has an invalid section specifier 'notes': mach-o section specifier requires a segment and section separated by a comma.
```
I'll include this somewhere. As I recall there were other restrictions as well but I don't recall what they were and they don't show up in just a build.